gammapy/data/tests/test_data_store.py

```diff
@@ -335,3 +335,5 @@ def test_data_store_no_events():
     for obs in observations:
         assert not obs.events
         assert not obs.gti
+
+    utils.CHECK_OVERLAPPING_TI = True
```

Comment:
Why is this needed ? Does the user have to know is this is needed or not ? can't we determine automatically if the check if needed ?

Reply:
The overlap catched by the check is of the order of 0.0001s for gtis whitin one observation, so I suspect the failure just comes from a precision error.

Reply:
events and GTIs should have the same accuracy. Have we made an error on the events time accuracy?